src/core/git/gitManager.ts

```diff
@@ -0,0 +1,147 @@
+import path from "node:path";
+import { type SimpleGit, type SimpleGitOptions, simpleGit } from "simple-git";
+import type { NewBranch, NewProject } from "../db/schema";
+import { getErrorMessage } from "../utils/error";
+import {
+  EmptyCommitMessageError,
+  GitError,
+  IsNotGitRepositoryError,
+} from "./errors";
+
+interface GitServiceConfig {
+  gitProvider?: typeof simpleGit;
+  repoPath?: string;
+  gitOptions?: Partial<SimpleGitOptions>;
+}
+
+const gitInstances = new Map<string, SimpleGit>();
+
+export function createGitService(config: GitServiceConfig = {}) {
+  const {
+    gitProvider: gitFactory = simpleGit,
+    repoPath = process.cwd(),
+    gitOptions = { trimmed: true },
+  } = config;
+
+  function getGitInstance(): SimpleGit {
+    const normalizedPath = path.resolve(repoPath);
+
+    if (!gitInstances.has(normalizedPath)) {
+      const newGitInstance = gitFactory(normalizedPath, gitOptions);
+      gitInstances.set(normalizedPath, newGitInstance);
+    }
+
+    return gitInstances.get(normalizedPath)!;
+  }
```

Comment:
How about this approach?

```suggestion
  function getGitInstance(): SimpleGit {
    const normalizedPath = path.resolve(repoPath);
    let instance = gitInstances.get(normalizedPath);
    
    if (!instance) {
      instance = gitFactory(normalizedPath, gitOptions);
      gitInstances.set(normalizedPath, instance);
    }
    
    return instance;
  }
```

Reply:
I like that it removes the get call at the end and it's easier to read.